public/app/plugins/datasource/loki/language_provider.ts

```diff
@@ -6,12 +6,13 @@ import { parseSelector, labelRegexp, selectorRegexp } from 'app/plugins/datasour
 import syntax from './syntax';
 
 // Types
-import { CompletionItem, LanguageProvider, TypeaheadInput, TypeaheadOutput, HistoryItem } from 'app/types/explore';
+import { LanguageProvider, HistoryItem } from 'app/types/explore';
```

Comment:
Is it possible to move `LanguageProvider` to grafana-ui as well? It should be helpful for external plugins when you implement your own language provider.

Reply:
or grafana-data? Any thoughts on this @torkelo ?

Reply:
just the interface?

``´
export abstract class LanguageProvider {
  datasource: DataSourceApi;
  request: (url: string, params?: any) => Promise<any>;
  /**
   * Returns startTask that resolves with a task list when main syntax is loaded.
   * Task list consists of secondary promises that load more detailed language features.
   */
  start: () => Promise<any[]>;
  startTask?: Promise<any[]>;
}
```

It cannot be in @grafana/data yet as DataSourceApi is in @grafana/ui

Reply:
or do you mean extract a base implementation of the language provider?